### DISCUSSION TITLE
libp2p Day 2023

### DISCUSSION THREAD

**Discussion by p-shahi** [libp2p/specs#591] [Announcements]
Hi everyone 👋,

The maintainers of go-libp2p, rust-libp2p, and js-libp2p libp2p at Protocol Labs are organizing libp2p Day as a part of Protocol Labs' [LabWeek '23](https://23.labweek.io/)!

- [When and Where ⏰ 🗺️](#when-and-where--️)
- [What ⛲️](#what-️)
- [Content 🗣️ 📣](#content-️-)
  - [Process for submitting extra content](#process-for-submitting-extra-content)
- [Schedule 🎞️](#schedule-️)
- [Registration](#registration)
- [Questions and Points of Contact](#questions-and-points-of-contact)
- [Need inspiration? 🤓](#need-inspiration-)

## When and Where ⏰ 🗺️
**libp2p Day will be held in Istanbul, Turkey on Wednesday, November 15 (Location: [Neuchatel Room, Swissotel](https://www.google.com/maps?sca_esv=579932655&output=search&q=Swiss%C3%B4tel+The+Bosphorus+Istanbul&source=lnms&entry=mc&sa=X&ved=2ahUKEwjd7IPltbCCAxV2O0QIHWyKDU8Q0pQJegQIDBAB)).**

## What ⛲️
libp2p Day 2023 is a watering hole for the libp2p community.  This is not a Protocol Labs dominated event (in fact, many of the PL maintainers will be unable to attend).  The organizers hope the libp2p community feels empowered to share anything ranging from recent wins, updates, current challenges, deep dives, demos, etc.  This can range from a quick lightning talk, show-and-tell demo, to a more thorough deep dive.  Anything is welcome - there is no prescribed formula here!

## Content & Schedule 🗣️ 📣
It is an all day event with pre-planned talks as well as off the cuff un-conference sessions!

LabWeek website: https://23.labweek.io/libp2p-day
The current list of talks is:

| Talk | Time | Speaker (github handle) | Speaker Name and Team | Slides | Recording |
| - | - | - | - | - | - |
| [libp2p 2023 Lookback](https://github.com/libp2p/specs/discussions/591#discussioncomment-7566342) | 10:15 AM - 10:25 AM | @biglep | Steve Loeppky, Protocol Labs | [Slides](https://drive.google.com/file/d/1bEikd6eH_RegEYWWTBwoIAnQ50JkUmRd/view?usp=drive_link) | [Video](https://youtu.be/5wi1cyG-xMQ?si=EVdUMwhN8T__m2xz) |
| [Onboarding more developers to libp2p with the universal connectivity app](https://github.com/libp2p/specs/discussions/591#discussioncomment-7546402) | 10:30 AM - 10:45 AM | @2color | Daniel Norman, Protocol Labs | [Slides](https://drive.google.com/file/d/1UIf6ee4Zx4gcmpHR7ikE10dAczoDmaD1/view?usp=drive_link) | [Video](https://youtu.be/9qE6JE_GN_U?si=Yry5r9gckRXdNwMa) |
| [Full Stack rust-libp2p apps, with wasm and WebRTC](https://github.com/libp2p/specs/discussions/591#discussioncomment-7567161) | 10:45 AM - 11:05 AM | @DougAnderson444 | Doug Anderson, PeerPiper | [Slides](https://github.com/DougAnderson444/libp2pday2023) | [Video](https://youtu.be/cROI6RMyQ70?si=iUubKLDQePmdlOPS) |
| [AutoNAT v2](https://github.com/libp2p/specs/discussions/591#discussioncomment-7557879) | 11:05 AM - 11:20 AM | @sukunrt  | Sukun T., Protocol Labs | [Slides](https://drive.google.com/file/d/1Mex_A7ZtYnoux_-HUDMxp3N4A4bd1qRL/view?usp=drive_link) | [Video](https://youtu.be/OP-tzl9jPxE?si=2iG0gePI9axtBG9t) |
| [libp2p in .NET](https://github.com/libp2p/specs/discussions/591#discussioncomment-7520457) | 11:20 AM - 11:40 AM | @flcl42 | Alexey Osipov, Nethermind | [Slides](https://drive.google.com/file/d/1r2gaFdFG9pjbZEcY3bFXiunWQPH6zmy2/view?usp=drive_link) | [Video](https://youtu.be/K4e2bFcjfM8?si=-A4XHVyBmTDOWhgx) |
| [Javascript libp2p v1.0!](https://github.com/libp2p/specs/discussions/591#discussioncomment-7518853) | 11:40 AM - 12:00 PM | @wemeetagain | Cayman Nava, ChainSafe Systems | [Slides](https://drive.google.com/file/d/1FObJreuKNCpctqAec8ZqSMcFdeRvM4do/view?usp=drive_link) | [Video](https://youtu.be/Q5VKGWHAM38?si=_0uOmXx9KNY7zkSB) |
| [Data Availability and Data Reconstruction over p2p](https://github.com/libp2p/specs/discussions/591#discussioncomment-7553082) | 12:00 PM - 12:20 PM | @Wondertan | Wondertan, Celestia | [Slides](https://drive.google.com/file/d/1Z-1tPj-GeiJ3nbgZ9XaY5qJUXYt4Mjg1/view?usp=drive_link) | [Video](https://www.youtube.com/watch?v=Vtki5CPoj8g) |
| **_LUNCH BREAK_** | 12:30 PM - 1:30 PM | | | | |
| [GossipLog: durability and eventual delivery for Pubsub](https://github.com/libp2p/specs/discussions/591#discussioncomment-7502619) | 1:30 PM - 1:50 PM | @joeltg | Joel G., Canvas.XYZ | [Slides](https://drive.google.com/file/d/1wK-a08Iso8H4kEtk0nOkFi-YIrNO-9-j/view?usp=drive_link) | [Video](https://youtu.be/X8nAdx1G-Cs?si=wH_JT7TaIodhJne5) |
| [Waku : Uncompromising web3 communication at scale](https://github.com/libp2p/specs/discussions/591#discussioncomment-7519574) | 1:50 PM - 2:10 PM | @hackyguru | Guru, Status.Im | [Slides](https://www.canva.com/design/DAF0Kdu9XaQ/MFwAYgHtMTzXDZ8ylvvwfQ/view?utm_content=DAF0Kdu9XaQ&utm_campaign=designshare&utm_medium=link&utm_source=editor) | [Video](https://youtu.be/AkAFLZdxCCc?si=MbVJ41mefzC72ysx) |
| ⛳️ Leveling up the libp2p Project w/ a Foundation and Public Goods Funding | 2:10 PM - 2:40 PM | @jbenet, @raulk  | Juan Benet and Raúl Kripalani, Protocol Labs| | [Video](https://www.youtube.com/watch?v=z1xyJpBp1is) |
| [libp2p 2024 Roadmap brainstorm](https://github.com/libp2p/specs/discussions/591#discussioncomment-7566216) | 2:40 PM - 3:10 PM | @biglep | Steve Loeppky, Protocol Labs | | |
| ⚡️ **_Unconf Sessions/Lightning Talks_** |  |  |  | | |
| ⚡️ [State of js-libp2p in the browser and mobile](https://github.com/libp2p/specs/discussions/591#discussioncomment-7527763) | 3:15 PM - 3:25 PM | @maschad | Chad Nehemiah, Protocol Labs | [Slides](https://drive.google.com/file/d/1YKeRsv1eYfzGEMsqElidpR8-Nd5ZLTYe/view?usp=sharing) | [Video](https://youtu.be/49p3VtTbXko?si=Dm8fCpveRYqTigNr) |
| ⚡️ [pcp - A Fully Decentralized Peer-to-Peer File Transfer Tool](https://github.com/libp2p/specs/discussions/591#discussioncomment-7554708) | 3:25 PM - 3:35 PM | @dennis-tra | Dennis Trautwein, Protocol Labs | | [Video](https://youtu.be/dThtI_Y3kpA?si=3mpjMtibVWduuwtY) |
| ⚡️ [How Kiwi News uses libp2p-gossipsub](https://github.com/libp2p/specs/discussions/591#discussioncomment-7566248) | 3:35 PM - 3:45 PM | @TimDaub  | Tim Daub, Kiwi News Protocol | [Slides](https://drive.google.com/file/d/1SryrWpN4tW-itOGkqW3RxyUK4ob5Xhjq/view?usp=drive_link) | [Video](https://youtu.be/1U7tXX2HxOk?si=qXW45zlcNKgxv1jp) |
| ~⚡️ [How Libp2p did the hard work for me.](https://github.com/libp2p/specs/discussions/591#discussioncomment-7546427)~ | 3:45 PM - 3:55 PM | @Jorropo | Jorropo, Protocol Labs|  | |
| ⚡️ [Nebula - A Network Agnostic DHT Crawler](https://github.com/libp2p/specs/discussions/591#discussioncomment-7565963) | 4:15 PM - 4:30 PM | @dennis-tra  | Dennis Trautwein, Protocol Labs | | [Video](https://youtu.be/QDgvCBDqNMc?si=G3pCd67F0dKQR0_D) |
| ⚡️ [Leveraging Kademlia DHT for Efficient Data Availability Sampling](https://github.com/libp2p/specs/discussions/591#discussioncomment-7575410) | 3:55 PM - 4:05 PM | @prabal-banerjee  | Prabal Banerjee, Avail | [Slides](https://drive.google.com/file/d/1LHp6dOEnKJ-XTNfgID7E8j32FtmseULd/view?usp=sharing) | [Video](https://www.youtube.com/watch?v=17Dj3QPwLJ0)|
| ⚡️  "Your Lightning Talk Here" | 4:15 PM - 5:00 PM | YOU the reader | | | |

### Process for submitting extra content

More content is welcome 🙏.  Please propose your topic as a reply to [[this GitHub Discussion post announcing libp2p Day](https://github.com/libp2p/specs/discussions/591)] following this format:

```markdown
# Talk title

## Description
A description of your presentation

## Estimated runtime
How long you think your presentation will be (2 - 30 minutes).

## About you
Anything you want to share about yourself, which organization you're a part of, how you use libp2p, etc.
```

Community members can 👍 discussion posts to indicate support for a talk they’d like to hear.

Organizers will ensure time is made for them.

## Registration 
You can register to attend here: https://23.labweek.io/libp2p-day

## Questions and Points of Contact 
Please feel free to reach out to me (@ me in this post or email me at prithvi.shahi@protocol.ai) if you'd like to attend and would like to present on a libp2p related topic!

Day-of logistics and event MCing will be handled by @BigLep .

## Need inspiration? 🤓
_You can read about last year's libp2p Day here: https://blog.libp2p.io/2022-11-22-libp2p-day-2022-recap/_

**Comment by p-shahi** [comment]
Hello 👋,

Wanted to announce to folks about changes to libp2p Day.

Due to recent geo-political events, a majority of the Protocol Labs libp2p maintainers across go-libp2p, rust-libp2p, and js-libp2p will not be in attendance for libp2p Day.

### What does this mean for libp2p Day? 🤔

We still want to make this a successful event.  With less talks this year from go-libp2p, js-libp2p, and rust-libp2p maintainers, we’re even more reliant on community-generated content.  This event will be what the community makes of it.

@BigLep will be on the ground helping with event logistics and MCing. Even though we have less maintainers, we’re still invested in making this happen.

Our goals are:

- get together and build relationships as implementers, spec contributors, and users of libp2p 🤝
- empower you, the libp2p community, to share anything ranging from recent wins, updates, current challenges, deep dives, demos, etc. 📈
- treat libp2p Day ‘23 as a watering hole to discuss the future of libp2p ⛲

### What does this mean for me?❓

We already have a tentative plan for talks (see the [[Content section of the libp2p Day announcement post](https://github.com/libp2p/specs/discussions/591)](https://github.com/libp2p/specs/discussions/591)).

We’d like to invite you to propose a session and present on a libp2p/p2p related topic! Remember, as a libp2p community member, you’re an equal participant!

This talk can range from a quick lightning talk or show-and-tell demo to a more thorough deep dive.

Anything is welcome - there is no prescribed formula here! Whatever it is, you’ll help make the day a success. 🎉

### How do I participate? 🚀

We suggest you propose your topic as a reply to the [[GitHub Discussion post announcing libp2p Day](https://github.com/libp2p/specs/discussions/591)](https://github.com/libp2p/specs/discussions/591).

You can use the following format:

```markdown
# Talk title

## Description
A description of your presentation
## Estimated runtime
How long you think your presentation will be (2 - 30 minutes).

## About you
Anything you want to share about yourself, which organization you're a part of, how you use libp2p, etc.
```

This is to encourage public discussions and do things transparently in the open.

Community members can 👍 discussion posts to indicate support for a talk they’d like to hear.

We as organizers can ensure time is made for them.

And of course, hallway talks are always welcome 😄

Once we have a good list of talks, I will organize them into time slots and update the original post.

Thank you for your continued support in all things libp2p and hope to see you at libp2p Day ‘23!

Best Regards,

Prithvi

**Comment by joeltg** [comment]
# Talk title

GossipLog: durability and eventual delivery for Pubsub

## Description

Decentralized pubsub protocols are optimized for low-latency real-time delivery, but don't have a robust way for peers to catch up on missing messages after long periods of downtime or network partitions. We can use a new type of merkleized database index called Prolly trees to let peers efficiently sync large sets of messages in persistent storage. We can even extend this approach to add causal ordering using logical clocks.

## Estimated runtime

15 minutes

## About you

Joel is an ex-PL researcher and co-founder of Canvas, a framework for building automatically decentralized applications.

**Reply by p-shahi** [reply]
Added to the list of talks!

**Reply by RangerMauve** [reply]
Cool! Are the slides and source code published somehwere? Excited to watch the recording

**Reply by p-shahi** [reply]
@RangerMauve The link to slides is in the original post. Recording should be up soon.

**Comment by wemeetagain** [comment]
## Talk title
Javascript Libp2p v1.0!

## Description
Js-libp2p is fast approaching its v1.0 release (which may be released on libp2p day or shortly thereafter).
I'll be sharing the big struggles and wins behind the past 2 years of js-libp2p development.
Also giving a glimpse of the priorities for the next 6 months.

## Estimated runtime
15-20m

## About you
I'm a protocol engineer working at [ChainSafe](https://chainsafe.io) and helping with js-libp2p maintenance.

**Reply by p-shahi** [reply]
Thanks, updated the table with link to this post!

**Comment by hackyguru** [comment]
# Talk title
Waku : Uncompromising web3 communication at scale

## Description
In this talk, we will delve into Waku and explore its role in enabling decentralised communications at scale, with a focus on its synergy with the libp2p network stack. Waku is used by @XMTP, RAILGUN and TheGraph for enabling various use cases.
## Estimated runtime
25min

## About you
Guru is an ex-PL and is currently the DevRel for Waku

**Reply by p-shahi** [reply]
Added to the list of talks!

**Comment by flcl42** [comment]
# libp2p in .NET

## Description

Key features and perspectives of the new libp2p implementation. An introduction for contributors and architecture overview.

## Estimated runtime

15min

## About you

Ethereum Core Dev at Nethermind

**Reply by p-shahi** [reply]
Thanks, updated the table with link to this post!

**Comment by maschad** [comment]
# Talk title

State of js-libp2p in the browser and mobile

## Description

Will explore the stability and performance of libp2p in the browser and update on js-libp2p on mobile

## Estimated runtime
10 mins

## About you
 
On the js-libp2p team

**Reply by p-shahi** [reply]
Thanks, updated the table with link to this post!

**Comment by 2color** [comment]
# Talk title

Onboarding more developers to libp2p with the universal connectivity app

## Description

p2p is hard. If you're unfamiliar with libp2p, it can be intimidating. The universal-connectivity example showcases how neatly libp2p abstracts ubiquitous peer-to-peer connectivity between multiple programming languages (Go, Rust, TypeScript) and runtimes (Web, native binary) using QUIC, WebRTC and WebTransport. 

But it's still missing some key parts like ambient peer discovery for it to be a forkable example that can also serve as an educational resource. This takes away from the potential magic it could offer. 

In this brief talk, I'll explore:
- Why universal-connectivity is so important for the libp2p project  
- How we can use it to onboard more developers to using libp2p
- Challenges with peer discovery and some of the explored solutions
- Additional use-cases the app could explore 

## Estimated runtime

15 minutes 

## About you

Developer Advocate for IPFS and tinkerer with libp2p

**Reply by p-shahi** [reply]
Thanks, updated the table with link to this post!

**Comment by Jorropo** [comment]
# Talk title
How Libp2p did the hard work for me.
## Description
20 lines of code app demo about how to use go-libp2p and what are the values libp2p implements for you.
## Estimated runtime
4~8 minutes.

**Reply by p-shahi** [reply]
Thanks, updated the table with link to this post!

**Comment by Wondertan** [comment]
# Talk title
Data Availability and Data Reconstruction over p2p

## Description
I will go over the DA problem, it's importance, existing solutions, and what has yet to be solved.

## Estimated runtime
15-20

## About you
Team Lead and Protocol Engineer at Celestia

**Reply by p-shahi** [reply]
Thanks, updated the table with link to this post!

**Reply by Wondertan** [reply]
@p-shahi, what's the policy on going over time? Now, I feel like my presentation will take longer

**Reply by Wondertan** [reply]
Also, are these going to be recorded?

**Reply by p-shahi** [reply]
@Wondertan it's ok if your presentation goes over a few minutes for the presentation and Q/A afterwards. We have enough buffer at the end of the day.
Yes they will be recorded

**Comment by dennis-tra** [comment]
# Talk Title

pcp - A Fully Decentralized Peer-to-Peer
File Transfer Tool

## Description

A quick lightning talk about [pcp](https://github.com/dennis-tra/pcp), how it works and where development is at.

## Estimated runtime

5 - 10m

## About you

Research/Software Engineer at Protocol Labs

**Reply by p-shahi** [reply]
Thanks, updated the table with link to this post!

**Comment by sukunrt** [comment]
# Talk title

## Description
AutoNAT v2

## Estimated runtime
10 - 15 minutes

## About you
I am implementing AutoNAT for go-libp2p

**Comment by Wondertan** [comment]
@p-shahi, formatting seems to be broken on the speaker table

**Reply by Wondertan** [reply]
![image](https://github.com/libp2p/specs/assets/22449364/cd098fc0-ff6c-4bdd-84a8-8d9caa7e6dfa)

**Reply by BigLep** [reply]
Thanks @Wondertan - that was my bad.  I'll fix now.  I had it fixed but looks like it didn't save.  Doh!

**Comment by BigLep** [comment]
We'll get the schedule filled in by EOD 2023-11-14, but note that we'll plan on starting this event around 10am.  (We'll also get https://23.labweek.io/libp2p-day updated).  This github discussion will be the source of truth for schedule.

**Comment by sukunrt** [comment]
@Jorropo is volunteering to do a workshop if we have time.

**Reply by BigLep** [reply]
@Jorropo @sukunrt : can you please provide more details with your thoughts for this?  I'm trying to gauge the relevance for the audience.  
Something from last year that had great engagement was a discussion of security bugs that were fixed in libp2p implementations (got engagement around "spot the problem here").

**Comment by dennis-tra** [comment]
# Talk Title

Nebula - A Network Agnostic DHT Crawler

## Description

We're going to attempt a live crawl of the entire IPFS Amino DHT and take a brief look at the data.

## Estimated runtime

5m

## About you

Research/Software Engineer at Protocol Labs

**Reply by BigLep** [reply]
You were added to the list - thanks @dennis-tra !

**Comment by BigLep** [comment]
# libp2p 2024 roadmap brainstorm

## Description
A collaborative time to brainstorm about functionality we hope to see in libp2p implementations or the libp2p community. 

Collaborative doc for doing this together will come.

References (but outdated) items from the past:

- https://github.com/libp2p/specs/blob/master/ROADMAP.md
- https://github.com/libp2p/go-libp2p/issues/1806
- https://github.com/libp2p/rust-libp2p/blob/master/ROADMAP.md
- https://github.com/libp2p/js-libp2p/blob/master/ROADMAP.md
- https://github.com/libp2p/test-plans/blob/master/ROADMAP.md


## Estimated runtime
30 minutes

## About you
I'm an engineering manager with the PL EngRes libp2p team which are the primary maintainers of go/js/rust libp2p.

**Reply by BigLep** [reply]
Collaborative doc for doing this brainstorm is: https://docs.google.com/document/d/1EkAh54qbluERM-u6k9UFe_vGfDyKn7zw2dxlIvnbeCs/edit

**Comment by TimDaub** [comment]
# How Kiwi News uses libp2p-gossipsub

## Description

hey,
I just learned about this event today, but I'd be interested in giving a talk. Similar to Farcaster, we're using set reconciliation to synchronize our "decentralized Hacker News" nodes using libp2p and particularly libp2p-gossipsub.

Repo: https://github.com/attestate/kiwistand
Product: https://news.kiwistand.com

I have a talk ready-made that I gave at protocol.berlin and on Monday at zusocial (slides: https://docs.google.com/presentation/d/18a4rllQZ7n-0UGPLgJjdd59j0wYbRjV9hjGreH44hyo/edit?usp=sharing). I could give that one again, but it is 40 minutes. Otherwise, I'd be happy to just do like 5 minutes to introduce my project and say a few words about how we use libp2p and, e.g. live-demo sync.


## Estimated runtime
5 minutes improv or 40minutes

## About you

Founder of Kiwi News Protocol. I blog at https://proofinprogress.com

**Reply by p-shahi** [reply]
@TimDaub Added your talk - let's shoot for 10 - 15 minutes if you can?

**Reply by TimDaub** [reply]
sounds good

**Comment by BigLep** [comment]
# libp2p 2023 Lookback

## Description
Reviewing some of the libp2p accomplishments from 2023 

## Estimated runtime
5 mins

## About you
I'm an engineering manager with the PL EngRes libp2p team which are the primary maintainers of go/js/rust libp2p.

**Comment by DougAnderson444** [comment]
# Full Stack rust-libp2p apps, with wasm and WebRTC

## Description
rust-libp2p now has browser WebRTC transport access, here are some insights and pointers on how to build wasm libp2p apps moving forward now that we have great wasm transports.

## Estimated runtime
15-20 minutes?

## About you
I wanted full stack rust-libp2p in WebRTC, now we have it. How should be build apps moving forward?

**Reply by BigLep** [reply]
Thanks @DougAnderson444 !  We've linked this post to the schedule above.

**Reply by DougAnderson444** [reply]
Great! I'll post the link to the slides as well: [https://github.com/DougAnderson444/libp2pday2023](https://github.com/DougAnderson444/libp2pday2023)

https://douganderson444.github.io/libp2pday2023

**Comment by BigLep** [comment]
FYI that there is a Telegram for this event - please feel free to join!  https://t.me/+s6yOhR_2m4JhNmMx

**Comment by prabal-banerjee** [comment]
# Leveraging Kademlia DHT for Efficient Data Availability Sampling

## Description
Discussion on some of the design choices and implementation challenges around Data Availability Sampling using libp2p in Avail light client network
[Leveraging_Kademlia_DHT_for_Efficient_Data_Availability_Sampling.pdf](https://github.com/libp2p/specs/files/13363274/Leveraging_Kademlia_DHT_for_Efficient_Data_Availability_Sampling.pdf)


## Estimated runtime
15-20 mins

## About you
Co-founder of Avail. Avail uses libp2p to implement efficient data availability sampling to secure the Avail network, which aims to form the essential base layer for rollups.

**Reply by BigLep** [reply]
@prabal-banerjee : you've been added to the schedule - thanks

**Comment by p-shahi** [comment]
Thank you everyone for participating! I'll close this discussion tomorrow.

**Reply by BigLep** [reply]
Before we close, will we add links to the videos and slides?

**Reply by p-shahi** [reply]
I added links to the slides and will update with recording links. Closing for now.

**Comment by p-shahi** [comment]
Updated the original post with links to videos on YouTube
